Veterinaria/veterinaria.ts

```diff
@@ -1,67 +1,78 @@
-import * as ReadlineSync from "readline-sync";
-
-class Veterinaria{
-    private id : number;
+import * as ReadlineSync from "readline-Sync";
+import  {Cliente} from "./cliente"
+import  {Paciente} from "./paciente"
+export class Veterinaria{
+    
+    protected id : number;
     private nombre : string;
     private direccion : string;
     private telefono : number;
-    //private clientes : Cliente[];
-    //private pacientes : Pacientes[];
-
-    constructor(id : number, nombre : string, direccion : string, telefono : number,/** clientes : Cliente[], pacientes : Pacientes[]**/) {
+    private clientes : Cliente[];
+    private pacientes : Paciente[];
+    
+    constructor(id : number, nombre : string, direccion : string, telefono : number) {
         this.id = id;
-        this.nombre = nombre;   
+        this.nombre = nombre;
         this.direccion = direccion;
         this.telefono = telefono;
-       // this.clientes = clientes;
-        //this.pacientes = pacientes;
-    }  
-
-    /**altaCliente(cliente:Cliente){
-        this.clientes.push(cliente);
-    }**/
-    
-    /**modificarCliente(cliente:Cliente){
-        for (let i = 0; i < this.clientes.length; i++) { 
-            this.clientes[i].nombre =  ReadlineSync.question('Nombre de Cliente');
-    }}**/
-
-    /**bajaCliente(cliente:Cliente){
-        this.clientes.splice(this.clientes.indexOf(cliente));
-    }**/
+        
+    }
 
-    /**altaPaciente(paciente:Pacientes){
-        this.pacientes.push(paciente);
-    }**/
 
-    //modificarPaciente(paciente:Pacientes){}
+    public generarId(): number {
+        return  this.id = Math.floor((Math.random() * 100) + 1);
```

Comment:
Este generarId lo tenemos que utilizar para los clientes (en este caso) cuando los tenemos que agregar, con el this estamos llamando al id de nuestra clase, o sea la veterinaria.
En todo caso podrías retornar directamente el `Math.floor(...)` y eso ya te retorna un numero aleatorio. Esta funcion podes utilizarla dentro del altaCliente cuando necesitas generar el id del cliente